packages/cli/src/cmds/agentInstaller/pythonAgentInstaller.ts

```diff
@@ -121,11 +121,29 @@ export class PipInstaller extends PythonInstaller {
   }
 
   async checkCurrentConfig(): Promise<void> {
-    const cmd = new CommandStruct(
-      'pip',
-      ['install', '-r', this.buildFile, '--dry-run'],
-      this.path
-    );
+    let commandArgs = ['install', '-r', this.buildFile];
+    let major: number;
+    let minor: number;
+    let patch: number;
+
+    try {
+      const pipVersionOutput = await getOutput('pip', ['--version'], this.path);
+      const VERSION_REGEX = /(?<major>\d+)\.(?<minor>\d+).(?<patch>\d+)/;
```

Comment:
This regex isn't a great match the version specifiers that pip uses: https://pypi.org/project/pip/#history .

We're already including the `semver` package in the cli. And, while pip's version specifiers don't follow semver, they're pretty close. You should be able to something like this (untested, make sure it works):
```
const supportsDryRun = semver.satisfies(semver.coerce(pipVersionOutput, {loose:true}, ">=22.2.0")
```

Reply:
Good catch, thank you. All of the pip versions have a major and a minor version, so I just changed the regex to look for a major and a minor version. I think this should work, but let me know if you still think I should use the `semver` package.

Reply:
I just noticed that certain pre-release versions have a `b` in them (i.e. `22.1b1` pre-release). This would still work with the current regex, giving `major` as `22` and `minor` as `1`.

Reply:
I still think you should switch to `semver`. We don't need to add code that duplicates its functionality.

Reply:
Okay, I'll change it